subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorParallelIntegrationTest.groovy

```diff
@@ -265,13 +285,33 @@ class WorkerExecutorParallelIntegrationTest extends AbstractWorkerExecutorIntegr
         blockingHttpServer.expectConcurrentExecution("workItem1")
 
         expect:
-        args("--max-workers=4")
+        args("--max-workers=3")
         succeeds("parallelWorkTask")
 
         and:
         output.contains("A failure occurred while executing work item 2")
     }
 
+    def "max workers is honored by parallel work"() {
+        given:
+        buildFile << """
+            task parallelWorkTask(type: MultipleWorkItemTask) {
+                doLast {
+                    6.times { i ->
+                        submitWorkItem("workItem\${i}")
+                        sleep 100
+                    }
+                }
+            }
+        """
+        blockingHttpServer.expectConcurrentExecution("workItem0", "workItem1", "workItem2")
```

Comment:
Agreed.  Thanks for pointing it out.  I added something to the test fixture to allow this sort of interaction with concurrent requests.  It's a little rough and could probably use some more polishing, but it works well enough for now.